pkg/kstatus/status/apply_status.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
+
+// ApplyStatus captures the apply status for a resource.
+//go:generate stringer -type=ApplyStatus -linecomment
+type ApplyStatus int
+
+const (
+	ApplyPending   ApplyStatus = iota // ApplyPending
+	ApplySucceeded                    // ApplySucceeded
+	ApplySkipped                      // ApplySkipped
+	ApplyFailed                       // ApplyFailed
+	PrunePending                      // PrunePending
+	PruneSucceeded                    // PruneSucceeded
+	PruneSkipped                      // PruneSkipped
+	PruneFailed                       // PruneFailed
+)
+
+// ReconcileStatus captures the reconcile status for a resource.
+//go:generate stringer -type=ReconcileStatus -linecomment
+type ReconcileStatus int
+
+const (
+	ReconcilePending   ReconcileStatus = iota // ReconcilePending
+	ReconcileSucceeded                        // ReconcileSucceeded
+	ReconcileSkipped                          // ReconcileSkipped
+	ReconcileTimeout                          // ReconcileTimeout
+	ReconcileFailed                           // ReconcileFailed
+)
+
+// ApplyReconcileStatus captures the apply and reconcile
+// status for a resource.
+type ApplyReconcileStatus struct {
```

Comment:
Why add this to kstatus?
The status enums already exist in the inventory package.
https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/inventory/types.go

Also, I thought we decided to use the 3 field API (Strategy, Actuation, Reconcile), instead of just 2 fields.

Reply:
Updated this PR. Please take another look.